src/ESM.sol

```diff
@@ -66,67 +52,26 @@ contract ESM is DSNote {
         require(z <= x);
     }
 
-    // -- admin --
-    function file(bytes32 job, address obj) external auth note {
-        if (job == "end") end = EndLike(obj);
-        if (job == "sun") sun = obj;
-    }
-
-    function file(bytes32 job, uint256 val) external auth note {
-        if (job == "cap") cap = val;
-    }
-
-    // -- state changes --
     function fire() external note {
-        require(state == START && full(), "esm/not-fireable");
+        require(!fired, "esm/already-fired");
+        require(sum >= min, "esm/min-not-reached");
 
         end.cage();
 
-        state = FIRED;
+        fired = true;
     }
 
-    function free() external auth note {
-        require(state == START || state == FIRED, "esm/not-freeable");
-
-        state = FREED;
-    }
-
-    function burn() external auth note {
-        require(state == START || state == FIRED, "esm/not-burnable");
-
-        sum   = 0;
-        state = BURNT;
-
-        bool ok = gem.transfer(address(sun), gem.balanceOf(address(this)));
-
-        require(ok, "esm/failed-transfer");
-    }
-
-    // -- user actions --
     function join(uint256 wad) external note {
```

Comment:
What do you think about having `require(!full())` here now. This could help to reduce the off chain complexity in the case where gov wants to mint new tokens for people.

Reply:
Honestly I don't know if it would really make a difference. If there's lots of small donations then it will be a PITA anyway, even just for the transactions below the threshold; if there's only a few large ones then they're likely coordinated and will likely use a multisig and do a `join+fire` from there.

Reply:
Yeah good point